test/testdata/desugar/forward_args.rb

```diff
@@ -18,6 +18,27 @@ def foo3(...)
   def foo4(*args, **kwargs, &block)
     T.unsafe(self).bar(*[1, 2, 3], *args, **kwargs, &block)
   end
+
+  def foo5(*)
+    [1,2,3].each do |*|
+                   # ^ error: Anonymous rest parameter
+      T.unsafe(self).p(*)
+    end
```

Comment:
The Syntax error is on line 25, on the use of anonymous *splat*, _not_ the anonymous rest parameter on the block.

```
unexpected `*` when the parent method is not forwarding
   3  def foo5(*)
   4    def foo6(*args)
   5      [1,2,3].each do |*|
>  7        T.unsafe(self).p(*)
   8      end
   9    end
  10  end
Untitled.rb:7: no anonymous rest parameter (SyntaxError)
```